adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java

```diff
@@ -154,6 +159,19 @@ public KuduEnumerable(final List<List<CalciteKuduPredicate>> predicates, final L
     this.calciteKuduTable = calciteKuduTable;
     this.filterFunction = filterFunction;
     this.isSingleObject = isSingleObject;
+
+    // groupFetchLimit calculates it's size based on offset. When offset is present,
+    // it needs to
+    // skip an equivalent number of unique group keys
+    if (offset > 0 && limit > 0) {
+      groupFetchLimit = limit + offset;
+    } else if (offset > 0) {
+      groupFetchLimit = offset;
+    } else if (limit > 0) {
```

Comment:
I don't think this is correct. I don't think you can set the `groupFetchLimit` if you only have an `offset`
```suggestion
    } else if (limit > 0) {
```